libopenage/renderer/opengl/shader_program.cpp

```diff
@@ -281,8 +276,66 @@ void GlShaderProgram::execute_with(const GlUniformInput *unif_in, const GlGeomet
 	}
 }
 
-std::unique_ptr<UniformInput> GlShaderProgram::new_unif_in() {
-	auto in = std::make_unique<GlUniformInput>();
+
+void GlShaderProgram::send_uniform(const GlUniformInput *unif_in) {
```

Comment:
This is already done in `execute_with`.

Reply:
Yes this is effectively the same program but without the sampler2d stuff as that is being handled by `texturemanager.cpp`